features/dd-sdk-android-flags/src/test/kotlin/com/datadog/android/flags/internal/JsonExtensionsTest.kt

```diff
@@ -0,0 +1,478 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.flags.internal
+
+import com.datadog.tools.unit.forge.BaseConfigurator
+import fr.xgouchet.elmyr.annotation.BoolForgery
+import fr.xgouchet.elmyr.annotation.DoubleForgery
+import fr.xgouchet.elmyr.annotation.IntForgery
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.json.JSONArray
+import org.json.JSONObject
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+
+@ExtendWith(ForgeExtension::class)
+@ForgeConfiguration(BaseConfigurator::class)
+internal class JsonExtensionsTest {
+
+    // region JSONObject.toMap()
+
+    @Test
+    fun `M convert primitives W toMap() {simple object}`(
+        @StringForgery key1: String,
+        @StringForgery value1: String,
+        @IntForgery value2: Int,
+        @BoolForgery boolVal: Boolean,
+        @DoubleForgery doubleVal: Double
+    ) {
+        // Given
+        val jsonObject = JSONObject().apply {
+            put(key1, value1)
+            put("number", value2)
+            put("bool", boolVal)
+            put("double", doubleVal)
+        }
+
+        // When
+        val result = jsonObject.toMap()
+
+        // Then
+        assertThat(result).hasSize(4)
+        assertThat(result[key1]).isEqualTo(value1)
+        assertThat(result["number"]).isEqualTo(value2)
+        assertThat(result["bool"]).isEqualTo(boolVal)
+        assertThat(result["double"]).isEqualTo(doubleVal)
```

Comment:
you may want to use `com.datadog.tools.unit.assertj.JsonObjectAssert` for such assertions

Reply:
thanks!